examples/bits.rs

```diff
@@ -1,98 +1,207 @@
+    for (num, row) in bits.chunks(16).enumerate() {
+        println!("Row {} bits: {:b}", num, row);
+        // Each chunk produced is a smaller `&BitSlice`, just like
+        // `&[bool].chunks` produces smaller `&[bool]`s, so we can draw it.
+        draw_row(&mut turtle, row);
+        // After each row, the turtle has to go back to the left edge.
+        turtle.set_x(left_edge);
     }
 
-    /// Set the bit at the given index to the given value
-    ///
-    /// Bits are set left to right, much like how an actual Vec might be indexed
-    ///
-    /// # Panics
-    /// Panics if index >= len()
-    pub fn set(&mut self, index: usize, value: bool) {
-        let byte_index = index / BITS;
-        let byte = &mut self.bytes[byte_index];
-        let bit_index = index % BITS;
-        // We want to make sure that bits are set left to right
-        // This depends on the platform's endianess
-        //TODO: Check this somehow
-        let bit_index = if cfg!(target_endian = "big") {
-            bit_index
-        } else {
-            BITS - 1 - bit_index
-        };
-        if value {
-            // Set the bit to 1 by performing a bitwise OR with each bit in the byte.
-            // `1 << bit_index` will make sure the bit_index bit is 1.
-            // 1 OR anything will always result in 1, so the bit will be set to 1.
-            *byte |= 1 << bit_index;
-        } else {
-            // Set the bit to 0 by performing a bitwise AND with each bit
-            // `!(1 << bit_index)` sets the bit_index bit to 0 and the other bits to 1.
-            // 1 AND x will always leave x as is.
-            // 0 AND x will set the bit to 0.
-            *byte &= !(1 << bit_index);
+    // Reader exercise:
+    //
+    // The IEEE-754 format for `f64` numbers separates them into three parts:
+    //
+    // 1. The sign marks whether the number is positive or negative: 1 bit
+    // 2. The exponent marks how far from zero the number is: 11 bits
+    // 3. The fraction describes the number: 52 bits.
+    //
+    // Using these widths (1 bit, 11 bits, 52 bits), the knowledge that
+    // `&BitSlice` is a normal Rust slice, and the API documentation for
+    // `std::iter::Iterator`, see if you can display each portion of an `f64`
+    // as its own row.
+    //
+    // Hints:
+    //
+    // - The variable `bits` is set up to view the entire number, from most
+    // significant bit to least.
+    // - You can get access to a structure that performs iteration by calling
+    //   `bits.iter()`.
+    // - You can use the `Iterator::by_ref` method to prevent `Iterator` adapter
+    //   functions from destroying the source iterator.
+    // - `&BitSlice` is an ordinary Rust slice, so you can use `[start .. end]`
+    //   range indexing to get smaller pieces of it.
+}
+
+/// Draw a row of bits on the screen.
+///
+/// This takes a reference to a turtle, which draws, and a reference to a slice
+/// of bits, which provides the data to draw.
```

Comment:
Maybe add:

```markdown
/// 
/// Note that this works whether we're going through the bits left to right (BigEndian)
/// or right to left (LittleEndian) because we assume that the turtle is going to start
/// on the correct side and be facing the correct way for this drawing to work.
```